src/models/profile.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+import mongoose from 'mongoose';
+import User from '../auth/user';
+import StatBlock from '../models/statBlock';
+
+const Schema = mongoose.Schema;
+
+const profileSchema = Schema({
+  user: { type: Schema.Types.ObjectId, ref: 'users' },
+  firstName: { type: String, required: true },
+  lastName: { type: String, required: true },
+  bio: { type: String, default: `Hello!` },
+  email: { type: String },
+  role: { type: String },
+
+});
+
+profileSchema.pre('save', function (next) {
+  let profileId = this._id;
+  let role = this.role;
+  let userId = this.user;
+  let statBlock = new StatBlock({ user: this.user });
+  statBlock.save();
+
+  User.findById(userId)
+    .then(user => {
+      if (!user) {
+        return Promise.reject('Sorry, unable to assign a profile to an invalid user');
+
+      } else {
+
+        if (role === 'coach') {
```

Comment:
The content of this if/elseif looks VERY similar. This should be an indication that things are not DRY and can be refactored. What happens when you add another role to RugbyScout? Just continually adding new else/if would not be ideal.